### ISSUE TITLE
Alternative scenarios for Map_blocks / Map_overlap / Overlap (feature requests)

### ISSUE THREAD

**Issue by helvince** [dask/community#428]
I am working with large scale/volume images and have had several use cases and observations that wonder slightly of the beaten path that I would like to share:

**1. Suboptimal inference on torch models**
I have noticed that loading/generating an image with cupy backend before converting to torch tensors is beneficial. But additionally, I found that inference using map_blocks/map_overlap on torch models is still not optimal. I observed that Increasing the chunk size, reshaping into a batch dimension, running inference and reshaping back is quite a bit faster.  E.g. running map_blocks on chunks of 1k x 1k is slower than taking larger chunks and batching them 8k x 1k -> 8 x 1k x 1k (inference) -> 8k x 1k.

Rechunking the data has its drawbacks of course, and when training a model it would be ideal to shuffle the batches too. It could therefore be a nice addition to include a batch_size argument to map_blocks. And a seed argument to divert from the natural order of the data to something random.

2. Shifted boundary condition map_overlap (for ViT inference)
The overlap and map overlap functions take the chunk and pad the sides with data from other chunks if available and if not, fills it in with constant/mirrored values. When running inference on a ViT, it would be more informative to shift and take more information from the actual image and trim accordingly.

1D Example: img = A B C.
Current chunks processed by map_overlap: 0Ab, aBc, bC0.
Alternatively, I would like Abb, aBc, bbC to be possible aswell, including (shifted) trimming.

**3. strided 'chunking' / inference**
Image segmentation using a pytorch model trained on lower resolution (Cell segmentation foundation model Cellposev3.1). Since such a CNN is effectively one convolutional filter, I decide to run it in a strided fashion.

# stupid but readable implementation
img = some_array
chunk1 = img[::2,::2]
chunk2 = img[1::2,::2]
chunk3 = img[::2,1::2]
chunk4 = img[1::2,1::2]
out[::2,::2] = CNN(chunk1)
etc

This can be done cleanly using reshaping, processing and reshaping back.
